internal/impl/redis/integration_test.go

```diff
@@ -100,6 +101,42 @@ input:
 		})
 	})
 
+	t.Run("stream-interpolated", func(t *testing.T) {
```

Comment:
It feels wrong to have this test separate, but otherwise I would have to change / extend all the other redis streams tests. Is there a better way? Same problem with the `getFn` approach.

Reply:
If you'd like to avoid adding an extra test, I'd probably just make a minor modification to `TestIntegrationRedis` and do `stream: ${! "stream" }-$ID` instead of `stream: stream-$ID`. This way, there's no need to add `StreamTestWithMetadata`, since the existing tests will cover it.

Reply:
Sounds like a good idea. Any suggestions on how to set custom metadata in messages through the test config? One option would be to add an additional `StreamTestOpt_` function and extend `StreamTestConfigVars`. Otherwise it seems that I would have to change the tests for the "streams" suite.

Reply:
Alright, fixed now. Removed the new test suite and used bloblang in the output processor to inject metadata in all streams tests. All are passing now.